### PR TITLE
Fixes #5763: Adjusts Mermaid graph theme

### LINKED ISSUE #5763: [Mermaid] Adjust graph theme to user defined theme (dark/light)
### Description

Currently, Mermaid diagrams are always displayed with the mermaid "normal" theme. This theme is not convenient if Zettlr is in dark theme. Of course, the theme can be set manually for a given chart, as follow:

````
```mermaid
%%{init: {'theme':'dark'}}%%
CODE
```
````
(see [doc](https://mermaid.js.org/config/theming.html))

### Proposed Changes

It would be great to have the following behavior:
- If Zettlr uses dark theme and nothing is set on the diagram: use mermaid `dark` theme.
- If Zettlr uses light theme and nothing is set on the diagram: use mermaid `normal` theme 
- If there is a theme setting on the diagram: do not override

I guess that this can be achieved by passing a global theme option when Zettlr calls the mermaid renderer - but no idea for the details.

### Caveats

I don't see caveats, except maybe if a user expects the normal theme even in dark mode? But this can be fixed on a per-graph level.

### Do you Wish to Attempt Implementing this Yourself?

No

### Zettlr Version

Stable (most recent version, 3.4.4), installed from the .deb

### Your Platform

- [ ] Windows
- [ ] macOS
- [x] Linux

### Operating System Version

Operating System: Linux Mint 12
KDE Plasma Version: 5.27.12
Qt Version: 5.15.13
Kernel Version: 6.8.0-59-generic (64-bit)

### Additional Information

Light theme and dark theme graph in dark mode Zettlr:

Image

### PR DESCRIPTION
**Pull request by Shafiia**
Closes #5763
## Description

This PR fixes the Mermaid graph theme rendering issue. The Mermaid diagrams' themes are set according to the light/dark theme of Zettlr when no theme setting exists on the diagram.
## Changes
render-mermaid.ts now includes a hasThemeConfig() function that detects if a diagram already has theme configuration. This ensures diagrams with theme config are rendered with their specified themes and diagrams without theme config use Zettlr's current theme. The IPC listener in render-mermaid.ts now re-renders existing charts when the theme changes, properly handling the theme detection logic for each chart.

## Additional information

Tested on: Ubuntu 24.04.1 LTS (noble) via WSL2 on Windows 11
Tested and working successfully.

### CODE REVIEW

**Comment by nathanlesage** [comment]
Looks fine; there are some merge conflicts, and I'll have to double-check this PR, as the inner workings of Mermaid always seem a bit complex, but I haven't found any direct problems other than that.

**Comment by Shafiia** [comment]
Thanks for reviewing the PR! Regarding the merge conflict, I think it might be due to inconsistent metadata values in CITATION.cff and the commit to update CITATION.cff (changing the version and release date) was done by an automated GitHub Action. I'll look into it. Please let me know if there's anything to modify.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 1]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
 import clickAndSelect from './click-and-select'
 import { trans } from '@common/i18n-renderer'
 
-// Always re-initialize mermaid as soon as the darkMode changes
+//Initialize mermaid without a global theme
+mermaid.initialize({ startOnLoad: false })
+
+//Helper function to detect if a diagram has theme configuration
+const hasThemeConfig = (graphData: string): boolean => {
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Please use function syntax here.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 2]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+  //Check for theme directive in various formats
+  const themePatterns = [
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts:42]: Please add a unit test for this function and include common edge cases, because this seems predestined for breaking depending on how lenient the Mermaid syntax is.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 3]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+    /%%{.*config.*theme.*}%%/i,   // %%{config: {theme: 'dark'}}%%
+    /%% theme:/i,                 // %% theme: dark
+    /theme\s*:/i                  // theme: dark (in config blocks)
+  ]
+  
+  return themePatterns.some(pattern => pattern.test(graphData))
+}
+
+//Helper function to get the fallback theme based on Zettlr's mode
+const getFallbackTheme = (): 'dark' | 'default' => {
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Again, please function syntax. Also, always add a docblock for each function.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 4]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+  const isDarkMode = window.config.get('darkMode') as boolean
+  return isDarkMode ? 'dark' : 'default'
+}
+
+//Helper function to render with appropriate theme
+const renderWithTheme = async (id: string, graphData: string): Promise<{ svg: string }> => {
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: ibid.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 5]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+}
+
+//Listen for theme changes via IPC
 const ipcRenderer = window.ipc
 ipcRenderer.on('config-provider', (event, { command, payload }) => {
   if (command === 'update' && payload === 'darkMode') {
-    const isDarkMode = window.config.get('darkMode') as boolean
-    const theme = isDarkMode ? 'dark' : 'default'
-    mermaid.initialize({ startOnLoad: false, theme })
+    console.log('Dark mode config changed, updating Mermaid charts without theme overrides')
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: No console logs

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 6]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+    
+    //Force re-render of existing mermaid charts
+    const existingCharts = document.querySelectorAll('.mermaid-chart')
+    console.log(`Found ${existingCharts.length} existing charts to update`)
+    
+    existingCharts.forEach((chart, index) => {
+      const chartElement = chart as HTMLElement
+      const graphData = chartElement.dataset.graph
+      
+      if (graphData != null && graphData !== '') {
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Can you simplify this if-condition?

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 7]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+        console.log(`Re-rendering chart ${index + 1}`)
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Remove console.logs

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 8]
```diff
@@ -21,19 +21,90 @@ import { type EditorState } from '@codemirror/state'
...
+        const id = `graphDiv${Date.now()}_${index}`
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Given how you changed the calculation of graph IDs — it might make sense to use the `hash`-function here to get a unique ID. Or, even better: Use UUID; the module is available in Zettlr's code base, and it should be fairly quick. But feel free to test it out and use your judgment.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts [hunk 9]
```diff
@@ -54,7 +125,7 @@ class MermaidWidget extends WidgetType {
 
     const id = `graphDiv${Date.now()}`
```
Comment (nathanlesage) [inline source/common/modules/markdown-editor/renderers/render-mermaid.ts]: See my note on ID generation above.

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts (hunk 6 above)
Comment (Shafiia) [reply source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Unfortunately, I couldn't yet, as the linter throws errors if nullish or empty cases aren’t explicitly handled. I'll look more into it

**Comment by Shafiia** [comment]
Thank you for the suggestions. I've pushed the edits. Let me know if there's anything else to update

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts (hunk 8 above)
Comment (Shafiia) [reply source/common/modules/markdown-editor/renderers/render-mermaid.ts]: Sounds good to me. I've used the hash function from Zettlr's code base and it seems to work well so far

File: source/common/modules/markdown-editor/renderers/render-mermaid.ts (hunk 2 above)
Comment (Shafiia) [reply source/common/modules/markdown-editor/renderers/render-mermaid.ts:42]: Thanks for pointing it out. I've edited the patterns and added a unit test file
